src/pages/home/grades/AllGradeEntryPage.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { useState } from 'react';
+import {
+  IonContent,
+  IonList,
+  IonModal,
+  IonPage,
+  IonToast,
+  IonButtons,
+  IonButton,
+  IonIcon,
+} from '@ionic/react';
+import { useHistory } from 'react-router-dom';
+import { useForm } from '@tanstack/react-form';
+import { add } from 'ionicons/icons';
+import ValidatedNumberInput from '@/components/Form/validated-number-input/validatedNumberInput';
+import Button from '@/components/Button/Button';
+import Header from '@/components/Header/Header';
+import GradeListItem from '@/components/List/GradeListItem';
+import FormField from '@/components/Form/FormField';
+import { Grade } from '@/db/entities';
+import {
+  useGrades,
+  useDeleteGrade,
+  useUpdateExamAndGrade,
+} from '@/hooks/queries';
+import {
+  validateGrade,
+  validateWeight,
+  percentageToDecimal,
+  decimalToPercentage,
+} from '@/utils/validation';
+import { useToast } from '@/hooks/useToast';
+import { Layout } from '@/components/Layout/Layout';
+import { Routes } from '@/routes';
+
+interface GradeFormData {
+  examName: string;
+  score: number;
+  weight: number;
+  date: string;
+  comment: string;
+}
+
+const AllGradeEntryPage: React.FC = () => {
```

Comment:
```suggestion
const GradesOverviewPage: React.FC = () => {
```

would probably be a better name.